qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1952,7 +1952,9 @@ def count_ops(self, *, recurse: bool = True):
         Returns:
             Mapping[str, int]: a mapping of operation names to the number of times it appears.
         """
-        if not recurse:
+        if not recurse or not {"for_loop", "while_loop", "if_else", "switch_case"}.intersection(
+            self._op_names
+        ):
```

Comment:
Let's consider moving this set, along with similar constructs in `dagcircuit.py:894`, `dagcircuit.py:958` and `transpiler/preset_passmanagers/common.py:56` into a shared `frozenset` in `qiskit.circuit.controlflow`.

Happy to do it in a follow-up if you prefer, though.

Reply:
I can do it in this PR it's small enough I'm not too worried about scope creep in this.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10788/commits/b83c73c38dac3a1126ef85ec8927d04a0539ad7a